### ISSUE TITLE
Maven Plugin: Add custom .pure files to configuration

### ISSUE THREAD

**Issue by idcmp** [pure4j/pure4j#6]
As a developer on a large project, I have huge swaths of code that are effectively immutable except for a single method (in my case a call to Apache's ToStringBuilder). I would very much like to be able to force ToStringBuilder to be considered Pure, so that I can gain the benefits of purity checking on the rest of the code and I can look into what  to do about ToStringBuilder separately.

It would be ideal to be able to tell the pure4j maven plugin of extra .pure files of my own creation. It would further be handy to have one central .pure files (or set of them) in a multi-module Maven build. This is not as effortful as it may sound - the maven-pmd-plugin does exactly this.

The maven-pmd-plugin supports the ability to specify a list of rulesets in its configuration. The ruleset itself may exist in a jar file. 

```
             <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-pmd-plugin</artifactId>
                    <executions>
                        <execution>
                            <id>pmd-check</id>
                            <goals>
                                <goal>check</goal>
                            </goals>
                        </execution>
                    </executions>
                    <configuration>
                        <rulesets>
                            <ruleset>${pmd.ruleset}</ruleset>
                        </rulesets>
                    </configuration>
                    <dependencies>
                        <dependency>
                            <groupId>org.example</groupId>
                            <artifactId>code-style-requirements</artifactId>
                            <version>${code-style-requirements.version}</version>
                        </dependency>
```
This allows an organization to build an artifact which lists the PMD rules it wants, put them in an Maven artifact and then have it versioned and depended upon by different teams.

Possibly a separate issue, but it would be really cool if the maven plugin dumped a results.pure into the target directory after it ran which listed anything it found that was NOT_PURE. This would give me something analyse.

**Comment by robmoffat** [comment]
For some reason, I've only just seen this, but yes, this makes a lot of sense.  I think this was probably what I had in mind in the first place, but haven't implemented.  I will make this configurable so that you can set it up as a file set from the maven plugin:  is this what you had in mind?

**Comment by idcmp** [comment]
Hi! No worries. It's close. If it's a file, then each module in a multi-module maven build needs to know where that file is (or each module needs to unpack the file from a dependency). In the PMD example up there, the plugin has a dependency on "code-style-requirements" jar file which is built as part of the project (or outside the project). This jar file has a file called "pmd-ruleset.xml" in it which the PMD plugin uses for its rules.

Since most maven projects are multimodule, being able to locate the file can be challenging if it's a file in a fileset.

**Comment by robmoffat** [comment]
Yes, I "fileset" was the wrong term, sorry.   Currently, the .pure files are picked up from the class path of the project being compiled.  I am thinking you should provide a list of names of other resources on the class path to pick up.  

For example, if you had a resource called `idcmp.pure` in your class path, you could list this in the maven config.  Sound ok?

**Comment by robmoffat** [comment]
Ok, I've just committed a change.  Please take a look when you have time, it should answer your problem.  Look at the pure4j-examples project.   In there:
. The `pom.xml` file now references an `example.pure` file (in `src/main/resources`)
. `org.pure4j.examples.purelist.SomePureClass` demonstrates use of this pure list.
